spirv_reflect.c

```diff
@@ -1314,7 +1417,8 @@ static SpvReflectResult ParseNames(SpvReflectPrvParser* p_parser)
   return SPV_REFLECT_RESULT_SUCCESS;
 }
 
-static SpvReflectResult ParseDecorations(SpvReflectPrvParser* p_parser)
+// builtin parsing should be done with shader module (if global) or entry point (if local)
+static SpvReflectResult ParseDecorations(SpvReflectShaderModule* p_module, SpvReflectPrvParser* p_parser)
```

Comment:
Add parameter for storing needed builtin info. One result_id is not guaranteed to have only one builtin decoration.